docs/hello_nextflow/04_hello_science.md

```diff
@@ -0,0 +1,763 @@
+
+_Before:_
+
+```groovy title="hello-gatk.nf" linenums="49"
+input:
+    path input_bam
+    path input_bam_index
+```
+
+_After:_
+
+```groovy title="hello-gatk.nf" linenums="49"
+input:
+    tuple path(input_bam), path(input_bam_index)
+```
+
+Of course, since we've now changed the shape of the inputs that `GATK_HAPLOTYPECALLER` expects, we need to update the process call accordingly in the workflow body.
+
+### 3.5. Update the call to GATK_HAPLOTYPECALLER in the workflow block
+
+We no longer need to provide the original `reads_ch` to the `GATK_HAPLOTYPECALLER` process, since the BAM file is now bundled (in the form of a symlink) into the channel output by `SAMTOOLS_INDEX`.
+
+As a result, we can simply delete that line.
+
+_Before:_
+
+```groovy title="hello-gatk.nf" linenums="84"
+GATK_HAPLOTYPECALLER(
+    reads_ch,
+    SAMTOOLS_INDEX.out,
+```
+
+_After:_
+
+```groovy title="hello-gatk.nf" linenums="84"
+GATK_HAPLOTYPECALLER(
+    SAMTOOLS_INDEX.out,
+```
+
+That is all the re-wiring that is necessary to solve the index mismatch problem.
+
+### 3.6. Run the workflow to verify it works correctly on all three samples every time
+
+Of course, the proof is in the pudding, so let's run the workflow again a few times to make sure this will work reliably going forward.
+
+```bash
+nextflow run hello-gatk.nf
+```
+
+This time (and every time) everything should run correctly:
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.02.0-edge
+
+ ┃ Launching `hello-gatk.nf` [special_goldstine] DSL2 - revision: 4cbbf6ea3e
+
+executor >  local (6)
+[d6/10c2c4] SAMTOOLS_INDEX (1)       | 3 of 3 ✔
+[88/1783aa] GATK_HAPLOTYPECALLER (2) | 3 of 3 ✔
+```
+
+If you'd like, you can use `.view()` again to peek at what the contents of the `` output channel looks like:
+
+```groovy title="hello-gatk.nf" linenums="92"
+SAMTOOLS_INDEX.out.view()
+```
+
+You'll see the channel contains the three expected tuples (filepaths truncated for readability).
+
+```console title="Output"
+[.../4c/e16099*/reads_son.bam, .../4c/e16099*/reads_son.bam.bai]
+[.../42/e70b8b*/reads_father.bam, .../42/e70b8b*/reads_father.bam.bai]
+[.../18/23b4bb*/reads_mother.bam, .../18/23b4bb*/reads_mother.bam.bai]
+```
+
+That will be much safer, going forward.
+
+### Takeaway
+
+You know how to make your workflow run on multiple samples (independently).
+
+### What's next?
+
+Make it easier to handle samples in bulk.
+
+---
+
+## 4. Make the workflow accept a text file containing a batch of input files
+
+A very common way to provide multiple data input files to a workflow is to do it with a text file containing the file paths. It can be as simple as a text list with one file path per line and nothing else, or the file can contain additional metadata, in which case it's often called a samplesheet.
```

Comment:
```suggestion
A very common way to provide multiple data input files to a workflow is to do it with a text file containing the file paths.
It can be as simple as a text list with one file path per line and nothing else, or the file can contain additional metadata, in which case it's often called a samplesheet.
```